### PR TITLE
Check tables referenced in materialized views

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java [hunk 1]
```diff
@@ -331,6 +346,41 @@ public void testReferencedTablesWithViews()
...
+    @Test
+    public void testReferencedTablesWithMaterializedViews()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("SELECT test_column FROM mock.default.test_materialized_view", 2);
+
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+
+        List<TableInfo> tables = event.getMetadata().getTables();
+        assertThat(tables).hasSize(2);
```
Comment (sopel39) [inline testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:358]: why 2?

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java [hunk 2]
```diff
@@ -331,6 +346,41 @@ public void testReferencedTablesWithViews()
...
+        TableInfo table = tables.get(0);
+        assertThat(table.getCatalog()).isEqualTo("tpch");
+        assertThat(table.getSchema()).isEqualTo("tiny");
+        assertThat(table.getAuthorization()).isEqualTo("user");
+        assertThat(table.isDirectlyReferenced()).isFalse();
```
Comment (sopel39) [inline testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:363]: assert table names too?

File: core/trino-main/src/test/java/io/trino/connector/MockConnector.java [hunk 3]
```diff
@@ -73,6 +73,7 @@
     private final Function<ConnectorSession, List<String>> listSchemaNames;
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/connector/MockConnector.java:73]: Please add tests.

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java [hunk 4]
```diff
@@ -115,6 +117,19 @@ protected QueryRunner createQueryRunner()
                             SchemaTableName viewName = new SchemaTableName("default", "test_view");
                             return ImmutableMap.of(viewName, definition);
                         })
+                        .withGetMaterializedViews((connectorSession, prefix) -> {
+                            ConnectorMaterializedViewDefinition definition = new ConnectorMaterializedViewDefinition(
+                                    "select nationkey as test_column from tpch.tiny.nation",
```
Comment (kokosing) [inline testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:122]: SELECT..AS...FROM...

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java (hunk 1 above)
Comment (Praveen2112) [reply testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:369]: Two tables are involved .. One is the materialized view table and other is the source table of materialized view

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java (hunk 2 above)
Comment (Praveen2112) [reply testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:375]: Added it.

**Comment by Praveen2112** [comment]
@kokosing , @sopel39 AC

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java (hunk 1 above)
Comment (sopel39) [reply testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:369]: Technically, one doesn't access MV table. It's just being analyzed.

**Review by sopel39 — APPROVED** [review]

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java (hunk 1 above)
Comment (Praveen2112) [reply testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:369]: `event.getMetadata().getTables()` gives the table details which are being analyzed...(so it gives us the MV tables and tables which is used by MV table)

File: core/trino-main/src/test/java/io/trino/connector/MockConnector.java [hunk 5]
```diff
@@ -328,6 +331,27 @@ public void createView(ConnectorSession session, SchemaTableName viewName, Conne
...
+        }
+
+        @Override
+        public ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession session, ConnectorTableHandle tableHandle, List<ConnectorTableHandle> sourceTableHandles)
+        {
+            return new MockConnectorInsertTableHandle(((MockConnectorTableHandle) tableHandle).getTableName());
+        }
+
+        @Override
+        public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(ConnectorSession session, ConnectorTableHandle tableHandle, ConnectorInsertTableHandle insertHandle, Collection<Slice> fragments, Collection<ComputedStatistics> computedStatistics, List<ConnectorTableHandle> sourceTableHandles)
```
Comment (skrzypo987) [inline core/trino-main/src/test/java/io/trino/connector/MockConnector.java:347]: Line seems too long. Maybe arg/line

**Review by skrzypo987 — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/connector/MockConnectorFactory.java [hunk 6]
```diff
@@ -269,6 +275,12 @@ public Builder withGetViews(BiFunction<ConnectorSession, SchemaTablePrefix, Map<
             return this;
         }
 
+        public Builder withGetMaterializedViews(BiFunction<ConnectorSession, SchemaTablePrefix, Map<SchemaTableName, ConnectorMaterializedViewDefinition>> getMaterializedViews)
+        {
+            this.getMaterializedViews = requireNonNull(getMaterializedViews, "getViews is null");
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/connector/MockConnectorFactory.java:280]: `getMaterializedViews is null`

**Review by sopel39 — APPROVED** [review]
lgtm % small comment

**Pull request merged**